atomicapp/providers/openshift.py

```diff
@@ -571,3 +622,71 @@ def _get_config_values(self):
         return (result["providerapi"],
                 result["access_token"],
                 result["namespace"])
+
+    def extract(self, image, src, dest, update=True):
+        """
+        Extract contents of a container image from 'src' in container
+        to 'dest' in host.
+
+        Args:
+            image (str): Name of container image
+            src (str): Source path in container
+            dest (str): Destination path in host
+            update (bool): Update existing destination, if True
+        """
+        cleaned_image_name = Utils.sanitizeName(image)
+        pod_name = '{}-{}'.format(cleaned_image_name, Utils.getUniqueUUID())
+        container_name = cleaned_image_name
+
+        # Pull (if needed) image and bring up a container from it
+        # with 'sleep 3600' entrypoint, just to extract content from it
+        artifact = {
+            'apiVersion': 'v1',
+            'kind': 'Pod',
+            'metadata': {
+                'name': pod_name
+            },
+            'spec': {
+                'containers': [
+                    {
+                        'image': image,
+                        'command': [
+                            'sleep',
+                            '3600'
+                        ],
+                        'imagePullPolicy': 'IfNotPresent',
+                        'name': container_name
+                    }
+                ],
+                'restartPolicy': 'Always'
+            }
+        }
+        self.oc.deploy(self._get_url(self.namespace, 'Pod'), artifact)
+        # FIXME: Pod takes some time to come up, so we wait for 3 seconds
+        # The best solution would be to watch for the pod to come to
+        # the running state
+        time.sleep(3)
```

Comment:
Yeah. We pretty much have to query for the state of the pod and make sure it is ready in this step. We'll have to add that functionality and update this before we merge.

In my testing I had to up the sleep value because it was failing.